tools/java/src/main/java/org/unicode/cldr/json/Ldml2JsonConverter.java

```diff
@@ -546,23 +547,16 @@ private int convertCldrItems(AtomicInteger readCount, int totalCount,
                             throw new IllegalArgumentException("empty xpath in " + filename + " section " + js.packageName + "/" + js.section);
                         }
                         if (type == RunType.rbnf) {
-                            item.setValue(item.getValue().replace('→', '>'));
-                            item.setValue(item.getValue().replace('←', '<'));
-                            if (item.getFullPath().contains("@value")) {
-                                int indexStart = item.getFullPath().indexOf("@value") + 8;
-                                int indexEnd = item.getFullPath().indexOf("]", indexStart) - 1;
-                                if (indexStart >= 0 && indexEnd >= 0 && indexEnd > indexStart) {
-                                    String sub = item.getFullPath().substring(indexStart, indexEnd);
-                                    /* System.out.println("sub: " + sub);
-                                    System.out.println("full: " + item.getFullPath());
-                                    System.out.println("val: " + item.getValue());*/
-                                    item.setFullPath(item.getFullPath().replace(sub, item.getValue()));
-                                    item.setFullPath(item.getFullPath().replaceAll("@value", "@" + sub));
-                                    //System.out.println("modifyfull: " + item.getFullPath());
-                                    item.setValue("");
-                                }
+                            XPathParts xpp = XPathParts.getFrozenInstance(item.getFullPath());
+                            final String sub = xpp.findAttributeValue("rbnfrule", "value");
+                            if(sub != null){
+                                xpp = xpp.cloneAsThawed();
+                                final String value = item.getValue();
+                                xpp.removeAttribute(-1, "value");
+                                xpp.addAttribute(sub, value);
+                                item.setFullPath(xpp.toString());
+                                item.setValue("");
                             }
-
                         }
                         // ADJUST ACCESS=PRIVATE/PUBLIC BASED ON ICU RULE -- START
                         if (type == RunType.rbnf) {
```

Comment:
@btangmu I completely missed that there are two "if(type == RunType.rbnf)" conditionals right after another. Will fix.

Reply:
I missed that too